### ISSUE TITLE
Auto battle not working on Nox

### ISSUE THREAD

**Issue by Nemesis6Ghoul** [29988122/Fate-Grand-Order_Lua#211]
Hello all, Thanks for all the efforts put into this, It's really going to make my nerofest experience much less painful than last year if the next engine update doesn't affect this. Moving on, I have installed the script and auto FP rolling works, Auto battle on the other hand does not. Whenever I try to run the script it gives me an empty 'caution' message from nox followed by a pop up of my settings and nothing else happens. I am not really sure what's wrong here, Any help would be appreciated. I am on NA by the way.

Screenshots:

![Screenshot_2019-08-16-07-59-21](https://user-images.githubusercontent.com/40915867/63146730-3b5b1700-bffc-11e9-9160-f7442bc075c3.png)
![Screenshot_2019-08-16-07-59-26](https://user-images.githubusercontent.com/40915867/63146738-401fcb00-bffc-11e9-84fe-22fd6d57669f.png)

After that nothing happens at all.

EDIT: As of now, Only autosupport and autorepeat don't work, If you can point me towards what I am doing wrong with the script, It'd be appreciated. I even turned the support selection process to 'any' and still nothing. Here's the file

https://dl.dropbox.com/s/pz2kugnh75w6hkk/FGO_EN_REGULAR.lua

**Comment by Kizuro** [comment]
I am embarrassed to admit this but I also got the problem on using NOX when running the script. Doing the battle itself does not result in problem, but when I the script trying to choose support servant, it stop there. I already set the AutoSupportSelection to "first" so the picking process should have pick the first servant instead of staying still. Would like some tips on fixing this issue.

**Comment by TryBane** [comment]
Have you made any adjustments on the FGO_NA_REGULAR.lua file? This is needed to use some of the more advanced features the script provides and might explain a reason why this isn't working as it should.

**Comment by Nemesis6Ghoul** [comment]
I was testing if it'd work straight out of the box first and then I was planning to make the adjustment one by one to see if they'd break anything. I looked at the FGO_NA_REGULAR.lua file and I saw nothing that needed to be adjusted to pick waver with lunchtime as support for training grounds as a first try. I even thought it'd be okay if i could just get it to work in the battle itself then figure out how to get the auto support working, But no dice. I even tried installing a 7.1 version of nox and re-installing the game with a fresh account to try it on but still the same issue. Right now, I am suspecting the daemon process since I have installed the boot daemon upon launch, It might be affecting things but last time i tried to do the daemon manually, It didn't work so well. Can the boot daemon cause such an issue?

**Comment by Nemesis6Ghoul** [comment]
Yup, Seems like installing Daemon manually worked. Gonna have to do it everytime nox launches and the issue would be solved. I will keep this thread open if you guys don't mind until I edit everything in the FGO_NA_REGULAR.lua file and see if they work just so I don't clutter the issues tab with posts.

**Comment by Nemesis6Ghoul** [comment]
Okay, I have ran a script for the 40AP berserker training ground and most features seem to be working, A few polishing tweaks here and there now that I understand how its logic process works but two things seem to be not working as of now, Support selection and auto repeat. I will check the file again and update this thread.

**Comment by TryBane** [comment]
Support selection and auto repeat?

The IsContinuing should be set as 0 for NA so make sure that is the case. As for Support Selection, make sure you have it set as "Preferred" and not "Manual" or "First"

**Comment by TryBane** [comment]
Also, do you have a notch in your device? Examples of devices that have a Notch include S10 and Oneplus 6T. This has caused issues that I've had to solve manually for friends and people who have DMd me on Discord.

**Comment by Nemesis6Ghoul** [comment]
Okay, So for Auto repeat, IsContinuing is set to 0 in the regular.lua file but for some reason, The emulator seems to run slower when the script is running. So there's actually a 7 seconds-ish interval between when a quest is finished and when the bot should be trying to initiate a second quest. And it seems like the code has that interval at 2.5 seconds. At least that's how I am understanding it, I might be wrong. But everytime i try to modify the regular.lua file while having a backup, I get this error.

![Screenshot_2019-08-21-16-44-37](https://user-images.githubusercontent.com/40915867/63446154-f9860280-c439-11e9-8595-d6d5a343204a.png)

As for support, I honestly have no idea, I have set it to multiple settings, This is the last one, And still nothing.

`--AutoSupportSelection
Support_SelectionMode = "preferred"
Support_SwipesPerUpdate = 10
Support_MaxUpdates = 6
Support_FallbackTo = "first"
Support_FriendsOnly = 0
Support_PreferredServants = "waver4.png"
Support_PreferredCEs = "*chaldea_lunchtime.png"`

I suspect it might also be something that relates to longer interval times when running the script since loading the support screen takes less time when i am playing manually.

**Comment by Nemesis6Ghoul** [comment]
As for the notches, I am using Nox, But my model number is HUAWEI MLA-L12, Not sure if it'd matter though.

**Comment by Nemesis6Ghoul** [comment]
Changed the API to DirectX and the game seem to be working at normal speed with the script running but still no support selection or auto repeat.

**Comment by potchy** [comment]
Syntax error: unexpected symbol near '燎:
https://github.com/29988122/Fate-Grand-Order_Lua/blob/master/README.md#syntax-error-unexpected-symbol-near-%E7%87%8E

Em qua, 21 de ago de 2019 às 13:58, Nemesis6Ghoul <notifications@github.com>
escreveu:

**Comment by TryBane** [comment]
Oh I see! I had completely forgot about this in the readme. Thank you @potchy! Anyways, give that a change and try again. Let me know if anything else doesn't work as intended

**Comment by TryBane** [comment]
Oh and once I finish with some new features I'm planning to implement a debug feature for specific scenario testing. Should help with some problem solving.

**Comment by Nemesis6Ghoul** [comment]
Everything seems to be working today for some reason. xD Running and repeating X-F reliably. I haven't tried auto refill yet because I am waiting on nerofest for that but thanks, Everything seems to be in order for now.

**Comment by TryBane** [comment]
That's interesting. Post another issue if any other issues pop up.

**Comment by Nemesis6Ghoul** [comment]
Well, I think it may have been something related to the resolution or the DPI, I edited the file again to create a script for my bond farming and then auto repeat worked but support selection didn't. I tried to modify the DPI a bit but it still didn't work, I changed the resolution from the resolution of my monitor to 1920x1080 and now it worked, Made some custom pictures, Added them to the support selection and now everything running fine. Thanks for accommodating my problems, I just thought it's something you guys should keep in mind in case anyone runs into this problem again. My own screen resolution is 1366x768, So there might be something to look at there.
